components/brave_search/renderer/brave_search_render_frame_observer.cc

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_search/renderer/brave_search_render_frame_observer.h"
+
+#include <string>
+#include <vector>
+
+#include "base/no_destructor.h"
+#include "brave/components/brave_search/renderer/brave_search_default_js_handler.h"
+#include "brave/components/brave_search/renderer/brave_search_renderer_utils.h"
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "url/gurl.h"
+
+namespace brave_search {
+
+BraveSearchRenderFrameObserver::BraveSearchRenderFrameObserver(
+    content::RenderFrame* render_frame,
+    int32_t world_id)
+    : RenderFrameObserver(render_frame), world_id_(world_id) {
+  native_javascript_handle_.reset(
+      new BraveSearchDefaultJSHandler(render_frame));
+}
+
+BraveSearchRenderFrameObserver::~BraveSearchRenderFrameObserver() {}
+
+void BraveSearchRenderFrameObserver::DidCreateScriptContext(
+    v8::Local<v8::Context> context,
+    int32_t world_id) {
+  if (!render_frame()->IsMainFrame() || world_id_ != world_id)
+    return;
+
+  GURL url =
+      url::Origin(render_frame()->GetWebFrame()->GetSecurityOrigin()).GetURL();
+
+  if (!url.is_valid() || !url.SchemeIsHTTPOrHTTPS() ||
```

Comment:
these checks are redundant since they are done in IsAllowedHost and also we don't allow http

Reply:
fixed via https://github.com/brave/brave-core/compare/ce5bad2..0c92e3644999fdf0b18efa432d5a07778dc2d7b7